deepspeed/module_inject/containers/llama.py

```diff
@@ -122,10 +122,16 @@ def __init__(self, client_module, inference=True):
             LLAMALayerPolicy._orig_layer_class = None
 
     def get_hidden_heads(self):
-        return self.client_module.self_attn.q_proj.weight.shape[1], \
-                self.client_module.self_attn.num_heads, \
-                self.client_module.input_layernorm.variance_epsilon, \
-                self.client_module.mlp.gate_proj.weight.shape[0]
+        try: # for zero stage 3
+            return self.client_module.self_attn.q_proj.weight.ds_shape[1], \
+            self.client_module.self_attn.num_heads, \
+            self.client_module.input_layernorm.variance_epsilon, \
+            self.client_module.mlp.gate_proj.weight.ds_shape[0]
+        except:
+            return self.client_module.self_attn.q_proj.weight.shape[1], \
+                    self.client_module.self_attn.num_heads, \
+                    self.client_module.input_layernorm.variance_epsilon, \
+                    self.client_module.mlp.gate_proj.weight.shape[0]
```

Comment:
It would be nicer with `temp variable + getattr`.

```suggestion
        q_proj_weight = self.client_module.self_attn.q_proj.weight
        gate_proj_weight = self.client_module.mlp.gate_proj.weight
        return getattr(q_proj_weight, "ds_shape", q_proj_weight.shape)[1], \
                self.client_module.self_attn.num_heads, \
                self.client_module.input_layernorm.variance_epsilon, \
                getattr(gate_proj_weight, "ds_shape", gate_proj_weight.shape)[0]
```